pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -36,8 +37,61 @@ def __init__(self, runtime: Runtime, version: str, assembly: str, publish: bool,
         self.logger = runtime.logger
         self.working_dir = self.runtime.working_dir
         self.fail_count_name = f'{version}-assembly.{assembly}.count'
+        self.job_run = self.runtime.get_job_run_url()
+
+    async def comment_on_assembly_pr(self, text_body):
+        """
+        Comment the link to this jenkins build on the assembly PR if it was triggered automatically
+        """
+        # Keeping in try-except so that job doesn't fail because of any error here
+        try:
+            base_repo = "openshift-eng"
+            repository = "ocp-build-data"
+            branch = self.doozer_data_gitref
+            token = os.environ.get('GITHUB_TOKEN')
+
+            pattern = r"github\.com/([^/]+)/"
+            match = re.search(pattern, self.data_path)
+
+            repo_owner = None
+            if match:
+                repo_owner = match.group(1)
+
+            api = GhApi(owner=base_repo, repo=repository, token=token)
+
+            # Head needs to have the repo name prepended for GhApi to fetch the correct one
+            head = f"{repo_owner}:{branch}"
+            # Find our assembly PR.
+            prs = api.pulls.list(head=head, state="open")
+
+            if len(prs) != 1:
+                self.logger.error(
+                    f"{len(prs)} PR(s) were found from the auto generated branch {branch}. We need only 1.")
```

Comment:
```suggestion
                    f"{len(prs)} PR(s) were found with head={head}. We need only 1.")
```

Reply:
fixed in https://github.com/openshift-eng/aos-cd-jobs/pull/3892/commits/352cf9e3ffb448773cf29b37daa7b23d4d10a4d3